support/util/expose.go

```diff
@@ -42,3 +42,18 @@ func UseDedicatedDNSForKASByHC(hc *hyperv1.HostedCluster) bool {
 		// and later is used to annotate the route so the external DNS controller can watch it.
 		apiServerService.Route != nil && apiServerService.Route.Hostname != ""
 }
+
+func ServiceExternalDNSHostname(hcp *hyperv1.HostedControlPlane, serviceType hyperv1.ServiceType) string {
+	service := ServicePublishingStrategyByTypeForHCP(hcp, serviceType)
+	if service != nil && IsPublicHCP(hcp) {
```

Comment:
nit: for readability I'd usually avoid having the whole function/big chuncks within and if block and return earlier instead. e.g:

``
if !IsPublicHCP(hcp) {
    return ""
}

core logic here
```